### PR TITLE
WIP: taskbar: add Meta+1..9 shortcuts to raise windows

### PR DESCRIPTION
**Pull request by technic**
Hi LXQt developers!
I have added shortcuts to raise applications with Meta+number keys. Actually the code is mostly copypaste from `plugin-desktopswitch` and it possibly doesn't work as expected with grouped applications.
For me this "windows7 like" dock feature is vital. So if you are interested in such contribution I will put my efforts to enhance the code further.

Also I am dreaming about the ability to pin launcher as well, but it is much harder to implement.

### CODE REVIEW

**Comment by paulolieuthier** [comment]
I'm closing this due to inactivity. @technic, if you get back to work on it, feel free to reopen. Thanks.

**Comment by technic** [comment]
I did the fix

**Comment by technic** [comment]
Sorry, I pushed my commit to https://github.com/technic/lxqt-panel/commits/master but it is not here, and I don't know how can I update or reopen pull request :(

**Comment by palinek** [comment]
Please rebase to current master...

**Comment by technic** [comment]
rebased

File: plugin-taskbar/lxqttaskbar.cpp [hunk 1]
```diff
@@ -54,6 +56,7 @@ using namespace LXQt;
 ************************************************/
 LXQtTaskBar::LXQtTaskBar(ILXQtPanelPlugin *plugin, QWidget *parent) :
     QFrame(parent),
+    m_pSignalMapper(new QSignalMapper(this)),
```
Comment (palinek) [inline plugin-taskbar/lxqttaskbar.cpp]: Please, try to be coherent with the naming convetion used in the scope of the file/class

File: plugin-taskbar/lxqttaskbar.cpp [hunk 2]
```diff
@@ -83,6 +86,9 @@ LXQtTaskBar::LXQtTaskBar(ILXQtPanelPlugin *plugin, QWidget *parent) :
     QTimer::singleShot(0, this, SLOT(settingsChanged()));
     setAcceptDrops(true);
 
+    connect (m_pSignalMapper, SIGNAL(mapped(int)), this, SLOT(activateTask(int)));
```
Comment (palinek) [inline plugin-taskbar/lxqttaskbar.cpp]: Please, use the new Qt5 signal/slot syntax whenever possible

Comment (technic) [inline plugin-taskbar/lxqttaskbar.cpp]: here static_cast is required. Do you prefer static_cast syntax?

Comment (technic) [inline plugin-taskbar/lxqttaskbar.cpp]: or should I use kind of mixed?
connect (m_pSignalMapper, SIGNAL(mapped(int)), this, &LXQtTaskBar::activateTask);

Sorry I am confused...

Comment (palinek) [inline plugin-taskbar/lxqttaskbar.cpp]: I persolnaly would prefer static_cast....

**Comment by technic** [comment]
Simple logic will not work, because in mLayout we have a list of task from different desktops. I have to loop through and consider only visible buttons.

**Comment by palinek** [comment]
> Simple logic will not work, because in mLayout we have a list of task from different desktops. I have to loop through and consider only visible buttons.

Ah... I forgot about that. Yes, you're right.

File: plugin-taskbar/lxqttaskbar.cpp (hunk 2 above)
Comment (paulolieuthier) [inline plugin-taskbar/lxqttaskbar.cpp]: Me too.

**Comment by technic** [comment]
sync to recent master

**Comment by palinek** [comment]
@agaida @pmattern 
Should/Can I merge this? ... as this is about adding new shortcuts and affecting all current LXQt users (as probably all of them are using `plugin-taskbar` in their panel)

**Comment by technic** [comment]
Hi, I will think of a switch box in panel properties.

**Comment by agaida** [comment]
@palinek - we can test and discuss if when @technic has finished his work

**Comment by palinek** [comment]
>  and yes, we should merge it as soon as possible if ready - and have a look at user reactions.

It is ready... I think @technic just wanted to make using the shortcuts configurable (on/off), right?

If we just want to see users reactions, then we don't need the on/off option...

**Comment by agaida** [comment]
good point, merge it

**Comment by palinek** [comment]
landed in @21cc1991, thanks

**Comment by pmattern** [comment]
Bit late to the party but nevertheless wanted to state that the feature is working very well both "bare metal" and in virtual machines here. Quite an improvement, IMO.

(I figure the only problem from a user's point of view may be to know what a "meta key" is in the first place and how to configure it under different conditions like various window managers, now that this key is playing a role under LXQt the first time. Different story, of course.)

**Pull request closed without merging**